File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int length = nums.length;
        int[] results = new int[2];
        HashMap<Integer, Integer> dictionary = new HashMap<>();
        for(int index = 0; index < length; index++){
            int currentTarget = target - nums[index];
            if(dictionary.containsKey(currentTarget)){
                int currentTargetIndex = dictionary.get(currentTarget);
                results[0] = (currentTargetIndex < index) ? currentTargetIndex : index;
                results[1] = (currentTargetIndex > index) ? currentTargetIndex : index;
            }
            dictionary.put(nums[index], index);
        }
        return results;
    }
}

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var map = {};
    for(var index = 0; index < nums.length; index++){
        var anotherTarget = target - nums[index];
        if(anotherTarget in map){
            return [map[anotherTarget], index];
        }
        map[nums[index]] = index;
    }
    return null;
};